lib/vector-core/src/config/log_schema.rs

```diff
@@ -32,6 +32,7 @@ where
 /// Components should use global LogShema returned by this function.
 /// The returned value can differ from LogSchema::default()
 /// which is unchanging.
+#[must_use]
```

Comment:
I suppose it's weird to call this function without using its return value, but I've more commonly seen `must_use` with `Result` or guard types like `Mutex` where ignoring the return value could mask bugs. Is there a particular reason you'd like to mark these all as `must_use`?

Reply:
This is caused by adding the `pedantic` clippy rules, many of which look valuable but this one seems questionable.

Reply:
Clippy said it was a candidate was my reasoning. If we don't want to mark it `must_use` we'll either need to allow this one to pass without the lint being satisfied or not apply the lint to the whole package. On the whole I like the work that `must_use` does so I'm willing to tolerate the occasional odd one, like this one.

Reply:
Removed in a4bf98e